src/Ads/AccountService.php

```diff
@@ -211,17 +217,82 @@ public function get_billing_status(): array {
 			return [ 'status' => $status ];
 		}
 
+		$billing_url = $this->options->get( OptionsInterface::ADS_BILLING_URL );
+
+		// Check if user has provided the access and ocid is present.
+		$connection_status = $this->container->get( Connection::class )->get_status();
+		$email             = $connection_status['email'] ?? '';
+		$has_access        = $this->container->get( Ads::class )->has_access( $email );
+		$ocid              = $this->options->get( OptionsInterface::ADS_ACCOUNT_OCID, null );
+
+		// Deep link.
+		if ( $has_access && $ocid ) {
+			$billing_url = add_query_arg(
+				[
+					'ocid' => $ocid,
+				],
+				'https://ads.google.com/aw/signup/payment'
+			);
+		}
```

Comment:
There are going to be cases where we weren't able to save / keep the OCID. For example when they connect an existing account (without billing info added).

So in any case where they do have access we need to send them to the page `https://ads.google.com/aw/signup/payment` adding the OCID to turn it into a deeplink when possible.

Reply:
Good point. In [these two commits](https://github.com/woocommerce/google-listings-and-ads/pull/2270/files/40148ebf445d209e2b95c2b029387b0d4adb8e89..d941c7e1517ceae44b181823b1def176e62cd344) I've updated this logic to always deep link to the payment page whenever we already have access. If we don't have an ocid value, it will fall back to `0` which will make you select an ads account first, but then will send you to the billing setup page.